### ISSUE TITLE
add glossary

### ISSUE THREAD

**Issue by bjornregnell** [lunduniversity/introprog#295]
It would be nice if we had a glossary in an appendix in a tabular format:
```
 termInSwedish | termInEnglish | short swedish explanation, preferrably a crisp one-liner | link to wikipedia or other resource
``` 
Ideas on requirements:

* The data should be in case classes in scala code that is compiled and thus type checked (compare how the quizes are done)

* There should be methods to generate html, markdown and latex tables similar to how plans are generated

* there should be a new sbt command genGlossary that generates the concept tables

* The concept quizes has one-liners that should be used for a default version of terms in the glossary. The quiz statements are not perfect for the purpose of a glossary and better explanations should be added over time, but they are good starting points. Thus: It would be nice to generate a key value table with 
  ```
  ConceptId("english term") -> Concept("swedish term", "explanation in Swedish", Some("Url"))  
  ```
  from the quizes and for all elements add them to our map of concepts if we don't yet have made a "hard-coded" mapping. Talk to me and I'll explain what I mean... 

The code for glossary generation should be inspired by:
* Quiz data here: https://github.com/lunduniversity/introprog/tree/master/quiz
* Sbt commands here: https://github.com/lunduniversity/introprog/blob/master/build.sbt
* Plan generation here: https://github.com/lunduniversity/introprog/tree/master/plan

**Comment by bjornregnell** [comment]
@??? This might be something for a project in the course?

**Comment by bjornregnell** [comment]
This is just a sketch:
```
case class ConceptID(id: String)  // use preferred english term as id

case class Concept(
  term: String, 
  explanation: String,
  link: Option[String],
  language: String = "sv", 
  synonyms: Vector[String] = Vector()
)
```
Things to consider:
* How general should we be in terms of languages? Should it be possible to have German terms as well?
* Should we allow english synonyms?
* How should we produce sorted lists? Sorted on English preferred term? English synonyms? Swedish synonyms?

**Comment by bjornregnell** [comment]
fixed by 5fb9539868c5941480f6e8704323c615516023ee
